core/services/vrf/listener_v2.go

```diff
@@ -210,13 +210,13 @@ func (lsn *listenerV2) processPendingVRFRequests() {
 	lsn.pruneConfirmedRequestCounts()
 }
 
-func MaybeSubtractReservedLink(l logger.Logger, db *gorm.DB, fromAddress common.Address, startBalance *big.Int) (*big.Int, error) {
+func MaybeSubtractReservedLink(l logger.Logger, db *sqlx.DB, fromAddress common.Address, startBalance *big.Int) (*big.Int, error) {
 	var reservedLink string
-	err := db.Raw(`SELECT SUM(CAST(meta->>'MaxLink' AS NUMERIC(78, 0))) 
+	err := db.Get(&reservedLink, `SELECT SUM(CAST(meta->>'MaxLink' AS NUMERIC(78, 0))) 
```

Comment:
I've just noticed this query... looks like it might be unindexed and potentially quite expensive.

@connorwstein did you benchmark this?

Reply:
I did not, but this query hasn't appeared to inhibit the high scale vrfv2 request testing that @Bwest981 has done though

Reply:
I can open a follow up ticket to investigate though

Reply:
I'd like to see EXPLAIN ANALYZE run on a large real world database for this

Reply:
Opened https://app.shortcut.com/chainlinklabs/story/21433/benchmark-and-optimize-maybesubtractreservedlink